openedx/core/lib/block_structure/block_structure.py

```diff
@@ -564,7 +564,7 @@ def remove_transformer_block_field(self, usage_key, transformer, key):
         """
         try:
             transformer_block_data = self.get_transformer_block_data(usage_key, transformer)
-            delattr(transformer_block_data, key)
+            del transformer_block_data.fields[key]
```

Comment:
The problem here is that the attribute *does* exist, but the code as-written does not delete it due to an `AttributeError`, and the no-op `except` is swallowing the error.